subt_ros/src/BridgeLogger.cc

```diff
@@ -211,6 +218,36 @@ void BridgeLogger::OnSensorMsg(const topic_tools::ShapeShifter::ConstPtr &_msg,
     return;
   }
 
+  // check heartbeat
```

Comment:
How about moving this heartbeat check to a separate callback that is triggered by a ROS timer? I think this code will work as long as some sensor data comes through. If all sensor data stops, then we won't get a heartbeat error.

Reply:
yep done. Moved heartbeat check into its own thread. 296de9e